src/shims/unix/linux/epoll.rs

```diff
@@ -378,30 +379,25 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         events_op: &OpTy<'tcx>,
         maxevents: &OpTy<'tcx>,
         timeout: &OpTy<'tcx>,
-        place: &MPlaceTy<'tcx>,
+        place: &'tcx MPlaceTy<'tcx>,
     ) -> InterpResult<'tcx> {
         let this = self.eval_context_mut();
 
-        let epfd = this.read_scalar(epfd)?.to_i32()?;
+        let epfd_value = this.read_scalar(epfd)?.to_i32()?;
         let maxevents = this.read_scalar(maxevents)?.to_i32()?;
         let event = this.deref_pointer_as(
             events_op,
             this.libc_array_ty_layout("epoll_event", maxevents.try_into().unwrap()),
         )?;
-        let timeout = this.read_scalar(timeout)?.to_i32()?;
+        let timeout = this.read_scalar(timeout)?.to_u32()?;
 
-        if epfd <= 0 {
+        if epfd_value <= 0 {
             let einval = this.eval_libc("EINVAL");
             this.set_last_error(einval)?;
             this.write_scalar(Scalar::from_i32(-1), place)?;
             return Ok(());
         }
-        // FIXME: Implement blocking support
-        if timeout != 0 {
-            throw_unsup_format!("epoll_wait: timeout value can only be 0");
-        }
-
-        let Some(epfd) = this.machine.fds.get_ref(epfd) else {
+        let Some(epfd) = this.machine.fds.get_ref(epfd_value) else {
```

Comment:
use `dup` here and move the `epfd` (after making it a weak) into the callback. This way you get the check for whether it was closed before waking up for free